src/android/toga_android/widgets/label.py

```diff
@@ -1,40 +1,26 @@
-from android.view import Gravity
-
 from travertino.size import at_least
 
-from toga.constants import (
-    LEFT_ALIGNED,
-    RIGHT_ALIGNED,
-    CENTER_ALIGNED,
-    JUSTIFIED_ALIGNED,
-    NATURAL_ALIGNED,
+from .base import Widget
+from ..libs.android_widgets import (
+    TextView,
+    View__MeasureSpec,
 )
 
 
-class TogaLabel:
-    # TODO: Extend `android.widget.TextView`. Provide app as `context`.
-    def __init__(self, context, interface):
-        self.interface = interface
-
-
-class Label:
+class Label(Widget):
     def create(self):
-        self.native = TogaLabel(self.app.native, self.interface)
+        self.native = TextView(self._native_activity)
         self.native.setSingleLine()
 
-    def set_alignment(self, value):
-        self.native.setGravity({
-                LEFT_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.LEFT,
-                RIGHT_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.RIGHT,
-                CENTER_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
-                JUSTIFIED_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
-                NATURAL_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
-            }[value])
-
     def set_text(self, value):
-        self.native.setText(self.interface._text)
+        self.native.setText(value)
 
     def rehint(self):
-        # print("REHINT label", self, self.native.getMeasuredWidth(), self.native.getMeasuredHeight())
-        self.interface.intrinsic.width = at_least(self.native.getMeasuredWidth() / self.app.device_scale)
-        self.interface.intrinsic.height = self.native.getMeasuredHeight() / self.app.device_scale
+        self.native.measure(
+            View__MeasureSpec.UNSPECIFIED, View__MeasureSpec.UNSPECIFIED
+        )
+        self.interface.intrinsic.width = at_least(self.native.getMeasuredWidth())
+        self.interface.intrinsic.height = self.native.getMeasuredHeight()
+
+    def set_alignment(self, value):
+        return super().set_alignment(value)
```

Comment:
This seems like busy work - is there a reason set_alignment is on the base widget, and not label itself?

Reply:
No real reason. The same code seemed to maybe-work for `Button`. I have this here to mollify the test suite; otherwise I'd remove it.

Reply:
Done! See latest commit. Per our conversation on Gitter direct messages, in Toga-land, only `Label` supports `set_alignment()`, so it's better to have it on `Label`.